### PR TITLE
Ensure all config maps and volumes are deleted when a k8s operator is deleted

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

When deleting a k8s application, some operator artifacts (config map and storage volume) were being left behind.

## QA steps

deploy a k8s charm
delete the application
kubectl to check that the operator volume and config maps are gone

### CODE REVIEW

**Comment by wallyworld** [comment]
[operator deletion leaves behind storage pv](https://trello.com/c/tkMvKwvY/485-operator-deletion-leaves-behind-storage-pv)

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -1718,6 +1725,10 @@ func operatorConfigMapName(appName string) string {
 	return operatorName(appName) + "-config"
 }
 
+func operatorConfigurationsConfigMapName(appName string) string {
+	return deploymentName(appName) + "-configurations-config"
```
Comment (ycliuhw) [inline caas/kubernetes/provider/k8s.go]: wondering what the difference between `operatorConfigurationsConfigMapName` and `operatorConfigMapName`

Comment (wallyworld) [reply caas/kubernetes/provider/k8s.go]: i think this gets created as part of the stateful set setup maybe? no sure, but it is there and is left behind so we ensure it is deleted

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**